src/Nethermind/Nethermind.Blockchain/BeaconBlockRoot/BeaconBlockRootHandler.cs

```diff
@@ -39,7 +41,14 @@ public class BeaconBlockRootHandler(ITransactionProcessor processor, IWorldState
 
         if (includeStorageCells)
         {
-            builder.AddStorage(block.Timestamp % 8191);
+            // https://eips.ethereum.org/EIPS/eip-4788
+            // Set the storage value at header.timestamp % HISTORY_BUFFER_LENGTH to be header.timestamp
+            ulong slotIndex = header.Timestamp % HistoryBufferLength;
+            UInt256 slot256 = slotIndex;
+            builder.AddStorage(in slot256);
+            // Set the storage value at header.timestamp % HISTORY_BUFFER_LENGTH + HISTORY_BUFFER_LENGTH to be calldata[0:32]
+            slot256 = slotIndex + HistoryBufferLength;
+            builder.AddStorage(in slot256);
```

Comment:
Spec doesn't mention Access Lists; no gas is charged 30M should be enough for x2 SSTOREs

Spec is actually more wild and says

> Clients may decide to omit an explicit EVM call and directly set the storage values. Note: While this is a valid optimization for Ethereum mainnet, it could be problematic on non-mainnet situations in case a different contract is used.

Reply:
@MarekM25 ?